File: examples/linreg/example_press.py
import numpy as np, pandas as pd
import statsmodels.api as sm
from itertools import accumulate
def cumsum_text(lst, sep="+"):
    return list(accumulate(lst, lambda a, b: f"{a}{sep}{b}"))
boston = r["MASS::Boston"].drop(columns=['chas', 'rad']) 
model  = cumsum_text(boston.columns[:-1].tolist())
def calculate_press(model, data, y):
    X = sm.add_constant(data[model.split('+')])  
    return sm.OLS(data[y], X).fit().get_influence().ess_press
press = {}
for modeli in model:
    press[modeli] = calculate_press(modeli, boston, 'medv')
print(press)
print(min(press, key=press.get))

File: examples/linreg/example_press.R
cumsum_text <- function(x, sep = "+") {
  Reduce(function(a, b) paste(a, b, sep = sep), x, accumulate = TRUE)
}
#
library("olsrr")
bhd   <- MASS::Boston[,-c(4,9)]
model <- cumsum_text(c("1", names(bhd)[-12]))
lmi   <- lm(medv~1, data=bhd)
press <- numeric(0)
for (modeli in model) {
  lmi  <- lm(as.formula(paste0("medv~", modeli)), data=bhd)
  press[modeli] <- ols_press(lmi)
}
press
which.min(press)
